### DISCUSSION TITLE
How to send midi messages to port opened by other software?

### DISCUSSION THREAD

**Discussion by TommyHHH** [mido/mido#438] [Q&A] [answered]
Hello!
I'm trying to send midi messages to port opened by other software but not succeed: I have to use `mido.open_input` or `mido.open_output` to get an opened Input/Output instance first.

Is there any solution for sending midi messages to port opened by other software?
Thanks a lot!

**Comment by rdoursenaud** [comment, marked as answer]
Unless your driver is multiclient which, in my experience is rare (Only RME comes to mind), you can’t open an already opened port.
You may be able to workaround that by creating a virtual port for the other software and merging its data with yours.
